segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataStore.java

```diff
@@ -167,6 +177,20 @@ public void close() {
      */
     protected abstract CompletableFuture<Void> createSegment(String segmentName, ArrayView segmentInfo, TimeoutTimer timer);
 
+    /**
+     * Creates a new Transient Segment with given name.
+     *
+     * @param segmentName The case-sensitive Segment Name.
+     * @param attributes  The initial attributes for the StreamSegment, if any.
+     * @param timeout     Timeout for the operation.
+     * @return A CompletableFuture that, when completed normally, will indicate the TransientSegment has been created.
+     */
+    private CompletableFuture<Void> createTransientSegment(String segmentName, Collection<AttributeUpdate> attributes, Duration timeout) {
+        Collection<AttributeUpdate> attrs = new ArrayList<>(attributes);
+        attrs.add(new AttributeUpdate(Attributes.CREATION_EPOCH, AttributeUpdateType.None, this.connector.containerMetadata.getContainerEpoch()));
+        return Futures.toVoid(submitAssignment(newSegment(segmentName, SegmentType.TRANSIENT_SEGMENT, attrs), false, timeout));
```

Comment:
I recommend `submitAssignmentWithRetry`. We may have reached the maximum number of active segments (there is such a limit ~25,000 per container) and that method will attempt to evict some if it needs to make room. If you don't invoke that, your method will immediately fail if the container is full.